ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -169,22 +196,172 @@ module Json = struct
     | _ ->
         [("event", `Null); ("session", `Null)]
 
+  let of_result obj msg =
+    match msg.msg_result with
+    | None ->
+        `Null
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
+        then
+          `O
+            [
+              ("type", `String "EventBatch")
+            ; ("func_partial_type", `String "EventBatch")
+            ]
+        else
+          let t', _ = string_of_ty_with_enums t in
+          `O
+            [
+              ("type", `String t')
+            ; ("func_partial_type", `String (func_name_suffix t))
+            ]
+
+  let of_params _obj_name params =
+    let name_internal name =
+      let name = name |> snake_to_camel |> String.uncapitalize_ascii in
+      match name with "type" -> "typeKey" | "interface" -> "inter" | _ -> name
+    in
+    let of_param (t, name, doc, type_name) =
+      [
+        ("session", `Bool (name = "session_id"))
+      ; ("type", `String t)
+      ; ("name", `String name)
+      ; ("name_internal", `String (name_internal name))
+      ; ("doc", `String doc)
+      ; ("func_partial_type", `String type_name)
+      ]
+    in
+    (* We use ',' to seprate params in Go function, we should ignore ',' before first param,
+       for example `func(a type1, b type2)` is wanted rather than `func(, a type1, b type2)`.
+    *)
+    let add_first = function
+      | head :: rest ->
+          let head = `O (("first", `Bool true) :: of_param head) in
+          let rest =
+            List.map
+              (fun item -> `O (("first", `Bool false) :: of_param item))
+              rest
+          in
+          head :: rest
+      | [] ->
+          []
+    in
+    params
+    |> List.map (fun p ->
+           let fp_type = func_name_suffix p.param_type in
```

Comment:
I think this part can be moved into the `of_param` so that the paramter of `of_param` is indeed a `param`.